pkgs/http/lib/src/base_request.dart

```diff
@@ -89,14 +91,36 @@ abstract class BaseRequest {
   bool _finalized = false;
 
   static final _tokenRE = RegExp(r"^[\w!#%&'*+\-.^`|~]+$");
+
   static String _validateMethod(String method) {
     if (!_tokenRE.hasMatch(method)) {
       throw ArgumentError.value(method, 'method', 'Not a valid method');
     }
     return method;
   }
 
-  BaseRequest(String method, this.url)
+  /// On upload progress.
+  ///
+  /// If defined, this [HttpProgress.handler] will be called when the upload
+  /// progress changes.
+  ///
+  /// To see the usage of the progress handler, see [HttpProgress].
+  ///
+  /// To see the progress of the download, use [downloadProgress].
+  final HttpProgress? uploadProgress;
```

Comment:
My first impression:

Adding new fields to a public class is definitely going to be breaking:
https://github.com/aloisdeniel/http_extensions/blob/122a3d85632a08e0341e4701bafd0e8780740fab/http_extensions_base_url/lib/src/request.dart#L4

Reply:
Do you mind if I add this feature by creating a new Request and Client(io & html) classes? In global functions (get, post etc), In this case if xProgress parameters are set, this new type of Client will be instantiated.